### PR TITLE
Header sync user

### PR DESCRIPTION
**Pull request by duranmla**
**What:**
Add member hub link or join union depending on user session

**Why:**
We need to be able to distinguish between an active session and a guest session

**How:**
_Use `userSynced` event from Header in order to detect current user session_

**Media:**
https://www.loom.com/share/d5b87d1c45e0406eaf9ab0e33ee9c8ea

**Extras:**
_Clean code over layout so it makes more readable and we can better maintain different logics_

**Open question:**
1. What would happen if the user has a membership but not a community account?
2. Maybe should we check for membership and the community account?

_In my local I can have the case where I have a community user but no a membership_

### CODE REVIEW

File: src/components/Layout/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    const header = document.querySelector('#dc-header');
+    header?.addEventListener('userSynced', handleUserSynced);
+
+    return () => {
+      window.removeEventListener('userSynced', handleUserSynced);
+    };
+  }, []);
+
+  useEffect(() => {
+    const extraLinks = user ? USER_LINKS : GUEST_LINKS;
```
Comment (duranmla) [inline src/components/Layout/Header.tsx:75]: Maybe I need to keep `membership?.id` check here.
```
const extraLinks = (membership?.id || user.id) ? USER_LINKS : GUEST_LINKS;
```

Comment (orlando) [reply src/components/Layout/Header.tsx:75]: mmm this shouldn't be the case anymore, a User should always have a Discourse account now.

This `membership.id` comes from the `/users/current` endpoint on the membership app right?

**Review by arathjz — APPROVED** [review]

File: src/components/Layout/Header.tsx (hunk 1 above)
Comment (duranmla) [reply src/components/Layout/Header.tsx:75]: Then I will bet it is only my local and we can go with the User's discourse account.
And yes, the `membership.id` comes from the endpoint you describe on the membership app.

_Should I merge then?_

Comment (orlando) [reply src/components/Layout/Header.tsx:75]: maybe we can have a quick call to review this? I think we can remove the functionality of making a request to the membership app to request for session

Comment (duranmla) [reply src/components/Layout/Header.tsx:75]: I am not requesting to membership to check for session. 🤔

**Pull request merged**